### PR TITLE
Guozhong/hold threads for gpu

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *Hold threads for GPU for MULTI:GPU,CPU*

### Tickets:
 - *ticket-89110*

### CODE REVIEW

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -495,6 +495,9 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     std::mutex load_mutex;
     std::vector<Task> loads;
     std::once_flag readNetworkFlag;
+    std::string devCPUName;
+    int nCPUCores = parallel_get_max_threads();
```
Comment (peterchen-intel) [inline src/plugins/auto/plugin.cpp:499]: Check the nCPUCores on TGL/ADL/NUMA(XEON) if they are as expected (number of HW Cores)

Comment (wgzintel) [reply src/plugins/auto/plugin.cpp]: parallel_get_max_threads() gets 8 CPU Cores from TGL(machine: RIL DUT050-DG1MRB),  gets 4 CPU Cores from ADL(machine: RIL DUT4361-ADLN),  gets 64 CPU Cores from Xeon(R) Gold 6346(machine: odt-xiping-sonomacreek-01). The result matches CPU(s) which shows in lscpu.

Comment (peterchen-intel) [reply src/plugins/auto/plugin.cpp]: https://ark.intel.com/content/www/us/en/ark/products/212457/intel-xeon-gold-6346-processor-36m-cache-3-10-ghz.html

File: src/plugins/auto/plugin.cpp [hunk 2]
```diff
@@ -528,6 +548,42 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
                                      1 /*single thread per stream*/,
                                      IStreamsExecutor::ThreadBindingType::NONE});
     executor->runAndWait(loads);
+
+    if (iterCPU != metaDevices.end()) {
+        int nCPUCores = parallel_get_max_threads();
```
Comment (peterchen-intel) [inline src/plugins/auto/plugin.cpp:553]: What is the replacement if OneTBB/TBB is not used? Check the MACRO for TBB/OneTBB

File: src/plugins/auto/plugin.cpp (hunk 1 above)
Comment (wgzintel) [reply src/plugins/auto/plugin.cpp]: getNumberOfCPUCores() 
gets 4 CPU Cores from TGL(machine: RIL DUT050-DG1MRB CPU:11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz, Socket(s):1, Core(s) per socket:4)
gets 6 CPU Cores from Alder Lake S (machine: RIL DUT103-ADLS  CPU:12th Gen Intel(R) Core(TM) i5-12400, Socket(s):1, Core(s) per socket:6), 
gets 32 CPU Cores from Xeon(R) Gold 6346(machine: odt-xiping-sonomacreek-01 CPU:Intel(R) Xeon(R) Gold 6346 CPU @ 3.10GHz, Socket(s):2, Core(s) per socket:16). 
The result matches CPU(s) which shows in cmd lscpu.

File: src/plugins/auto/plugin.cpp (hunk 2 above)
Comment (wgzintel) [reply src/plugins/auto/plugin.cpp:553]: use getNumberOfCPUCores() to get CPU Cores

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**